### PR TITLE
TECH-467: Update Cucumber Test POST: /data/registryname/versionnumber…

### PR DESCRIPTION
**Pull request by mdulko**
### **Description**
Update cucumber test for POST /data/registryname/versionnumber/exists
Make sure tests contain

- smoke tests
- sanity performance tests
- no hard-coded data
- header tests
- etc. that are described in https://govstack-global.atlassian.net/wiki/spaces/GH/pages/157712385/Gherkin+file+template

Acceptance criteria:

- test is updated
- test match with accepted template

Ticket: https://govstack-global.atlassian.net/browse/TECH-467

### CODE REVIEW

File: test/openAPI/features/data_exist.feature [hunk 1]
```diff
@@ -1,19 +1,37 @@
...
 
-  Scenario: The user receives a message that the record exists in the Digital Registries database
-    Given The user wants to check if a record exists in the Digital Registries database
-    When The user sends a valid request to check if a record exists in the database
-    Then The process of receiving a message that a record exists in the database completes successfully
-    And The user receives the information that a record exists in the database
+  @smoke
+    Scenario: Successfully receives a message that the record exists in database
+    Given get the database users information of Digital Registries
+    When POST request with given path params "registryname" as registryname and "111" as versionnumber
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/data_exist.feature:7]: specify what is called. we can repeat it in .feature file, but if we do so repeating code for it in .js file will compile with errors.
So better make the text more unique

Comment (mgrochalskisoldevelo) [reply test/openAPI/features/data_exist.feature:7]: + for each given when then it works like that

File: test/openAPI/features/data_exist.feature [hunk 2]
```diff
@@ -1,19 +1,37 @@
...
+@method=GET @endpoint=/data/{registryname}/{versionnumber}/exists
+Feature: API endpoint that allows users to check if an entry exists in the database.
 
-  Scenario: The user receives a message that the record exists in the Digital Registries database
-    Given The user wants to check if a record exists in the Digital Registries database
-    When The user sends a valid request to check if a record exists in the database
-    Then The process of receiving a message that a record exists in the database completes successfully
-    And The user receives the information that a record exists in the database
+  @smoke
+    Scenario: Successfully receives a message that the record exists in database
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/data_exist.feature:5]:
```suggestion
    Scenario: Successfully receives a message that the record exists in database smoke type test
```

File: test/openAPI/features/data_exist.feature [hunk 3]
```diff
@@ -1,19 +1,37 @@
...
+    And response should have content-type: application/json header
+    And response should match json schema
 
-  Scenario: The user receives the message that the record does not exist in the Digital Registries database
-    Given The user wants to check if the record exists in the Digital Registries database
-    When The user sends a valid request to check if a record exists in the database
-    Then The process of receiving a message that a record exists in the database completes successfully
-    And The user receives the information that the record does not exist in the database
+  @positive @unit
+    Scenario: Successfully receive status value true for existing record
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/data_exist.feature:16]:
```suggestion
    Scenario: Successfully receives a message that the record exists in database
```

File: test/openAPI/features/support/data_exist.js [hunk 4]
```diff
@@ -1,97 +1,104 @@
...
+      })
+  );
+
+When(
+  /^given body "([^"]*)" as ID and "([^"]*)" as FirstName and "([^"]*)" as LastName and "([^"]*)" as BirthCertificateID$/,
+  (ID, FirstName, LastName, BirthCertificateID) =>
     specDataExist
-      .post(`${baseUrl}`)
-      .withHeaders(`${header.key}`, `${header.value}`)
       .withBody({
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/support/data_exist.js:44]:
```suggestion
      .withJson({
```

File: test/openAPI/features/support/data_exist.js [hunk 5]
```diff
@@ -1,97 +1,104 @@
...
-    })
+      chai
+        .expect(specDataExist._response.json.answer.status).to.be.true
 );
 
-Then('The operation returns an error due to an invalid request', async () => {
-  await specDataExist.toss();
-  specDataExist.response().should.have.status(400);
-});
+Then(
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/support/data_exist.js:95]: missing scenario description above + comment about others given,when,then

**Review by mgrochalskisoldevelo — CHANGES_REQUESTED** [review]

File: test/openAPI/features/data_exist.feature [hunk 6]
```diff
@@ -1,19 +1,37 @@
...
+    And response should have content-type: application/json header
+    And response should match json schema
+    And response should return status true for existing record
 
-  Scenario: The user is unable to verify that the record exists in the Digital Registries database due to an invalid request
-    Given The user wants to check if a record exists in the Digital Registries database
-    When The user sends an invalid request to check if the record exists in the database
-    Then The operation returns an error due to an invalid request
+  @positive @unit
+    Scenario: Successfully receives a message that the record not exists in database
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/data_exist.feature:28]: This scenario could be negative regression unit test.
Please add **Examples** data table and provide more examples. I guess we can use different name only, the rest could stay the same

File: test/openAPI/features/data_exist.feature [hunk 7]
```diff
@@ -1,19 +1,37 @@
...
+    And response should have content-type: application/json header
+    And response should match json schema
 
-  Scenario: The user receives the message that the record does not exist in the Digital Registries database
-    Given The user wants to check if the record exists in the Digital Registries database
-    When The user sends a valid request to check if a record exists in the database
-    Then The process of receiving a message that a record exists in the database completes successfully
-    And The user receives the information that the record does not exist in the database
+  @positive @unit
+    Scenario: Successfully receives a message that the record exists in database
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/data_exist.feature:16]: This test checks basically the same things as smoke test. I propose to add @unit and @positive tags to unit test.
Also add last And step to smoke test and remove the scenario

**Review by mgrochalskisoldevelo — CHANGES_REQUESTED** [review]

**Review by KarolinaKopacz — APPROVED** [review]

**Review by mgrochalskisoldevelo — APPROVED** [review]
lgtm

**Pull request merged**